### PR TITLE
Fix error when creating new key-pair (git)

### PR DESCRIPTION
**Pull request by Sm1th**
Now you can upload public keys to your heart's desire.

### CODE REVIEW

**Comment by reidka** [comment]
This looks good.  If I don't paste anything in or browse to a file, I get an error:
'' is not a valid SSH key string
Extracted source (around line #82):

    # Check to see if an individual repo exists for this user
    key = Gitolite::SSHKey.from_file(_path)

    ga_repo.add_key(key)

Also, the key file doesn't get deleted if I delete the key from the web interface.  Is that for another PR?

**Comment by Sm1th** [comment]
@reidka I will fix these in two separate pull requests

**Comment by Sm1th** [comment]
@reidka What should happen when a user inputs an invalid SSH key (such as an empty string)? Should there be an error message flashed?

**Comment by reidka** [comment]
@Sm1th Yes, an error message would be helpful.

File: app/controllers/key_pairs_controller.rb [hunk 1]
```diff
@@ -114,47 +115,53 @@ def create
...
-        format.html { render action: 'new' }
-        format.json do
-          render json: @key_pair.errors,
-                 status: :unprocessable_entity
+      end
+    else # if type and/or blob are nil
+      respond_to do |format|
+        format.html do
+          redirect_to :back,
+                      flash: { error: 'Key pair was not created: Invalid key.' }
```
Comment (reidka) [inline app/controllers/key_pairs_controller.rb]: Two comments:  First, please set up the message using flash_message (from helpers/application_helper.rb).  We are trying to move towards using this helper function more.  Second, the text of the message itself should go in the language files.  You'll need to add a key for it.

File: app/controllers/key_pairs_controller.rb [hunk 2]
```diff
@@ -192,7 +199,10 @@ def destroy
     @key_pair.destroy
 
     respond_to do |format|
-      format.html { redirect_to key_pairs_url }
+      format.html do
+        redirect_to key_pairs_path,
+                    notice: 'Key pair was successfully removed.'
```
Comment (reidka) [inline app/controllers/key_pairs_controller.rb]: As above, the message should go in the language file.

**Comment by Sm1th** [comment]
This pull request also includes fixes for issue #2455

**Comment by reidka** [comment]
@Sm1th  Looks good.  Please add the Portuguese language keys, and then I will merge it.  (You can just use google translate for place holder strings.)

**Comment by Sm1th** [comment]
@reidka Done :)

**Pull request merged**